docs/source/caching/programmatic-cache-keys.mdx

```diff
@@ -0,0 +1,188 @@
+---
+title: Programmatic Cache Keys
+---
+
+In situations where [declarative cache IDs](./cache-key-resolution) don't fit your use case, you can programmatically generate cache IDs for object types in your normalized cache.
+
+To configure cache keys programmatically, you can alter the [`SchemaConfiguration.cacheKeyInfo(for type:object:)`](https://www.apollographql.com/docs/ios/docc/documentation/apolloapi/schemaconfiguration/cachekeyinfo(for:object:)) method to return a `CacheKeyInfo` object.
+
+> **Note:** Programmatic cache key usage will take precedence over declarative cache keys. If you return a `CacheKeyInfo` for an object, it will be used to compute that object's cache key and any relevant `@typePolicy` will be ignored. Returning `nil` from this function will still fall back to using declarative cache keys.
```

Comment:
```suggestion
<Note>

Programmatic cache key usage will take precedence over declarative cache keys. If you return a `CacheKeyInfo` for an object, it will be used to compute that object's cache key and any relevant `@typePolicy` will be ignored. Returning `nil` from this function will still fall back to using declarative cache keys.

</Note>
```